python/lsst/daf/butler/registry/bridge/monolithic.py

```diff
@@ -249,6 +252,10 @@ def join_records(
         with self._db.query(info_in_trash) as sql_result:
             rows = [dict(row, datastore_name=self.datastoreName) for row in sql_result.mappings()]
 
+        # Only keep rows that match the external override selection.
+        if selected_ids is not None:
+            rows = [row for row in rows if row["dataset_id"] in selected_ids]
```

Comment:
I am worried that this still leaves a chance for a deadlock in case one uses `admin empty-trash` concurrently with regular removal. We probably need to use explicit locking to avoid that. What I would do is something like (line 248 above):

    info_in_trash = join_records(records_table._table.select(), self._tables.dataset_location_trash)
    if selected_ids:
        info_in_trash = info_in_trash.where(self._tables.dataset_location_trash.columns["dataset_id"].in_(selected_ids))
    info_in_trash = info_in_trash.with_for_update(skip_locked=True)

And I think that we want similar `.where()` for `items_in_trash` below too.